server/src/test/java/GroceryFamily/GroceryMom/api/ProductAPITest.java

```diff
@@ -0,0 +1,150 @@
+package GroceryFamily.GroceryMom.api;
+
+import GroceryFamily.GroceryElders.domain.Page;
+import GroceryFamily.GroceryElders.domain.Price;
+import GroceryFamily.GroceryElders.domain.Product;
+import GroceryFamily.GroceryMom.service.ProductService;
+import GroceryFamily.GroceryMom.service.exception.ProductNotFoundException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.client.HttpServerErrorException;
+
+
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+class ProductAPITest {
+
+    private MockMvc mockMvc;
+    @InjectMocks
+    private ProductAPI productAPI;
+    @Mock
+    private ProductService service;
+    private List<Product> products;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        this.mockMvc = MockMvcBuilders.standaloneSetup(productAPI).build();
+
+        Set<Price> prices = new HashSet<>();
+        Price price = Price.builder()
+                .unit("pc")
+                .currency("aden")
+                .amount(new BigDecimal(100))
+                .build();
+        prices.add(price);
+        Product product = Product.builder()
+                .namespace("zoo.buy")
+                .code("32133")
+                .name("Tiger 100 kg")
+                .prices(prices)
+                .build();
+        Product product1 = Product.builder()
+                .namespace("zoo.gou")
+                .code("42123")
+                .name("Rock 100 kg")
+                .prices(prices)
+                .build();
+        this.products = new ArrayList<>();
+        products.add(product);
+        products.add(product1);
```

Comment:
No need to prepare test set here. Just create all the necessary fixtures in place.